src/Symfony/Component/AssetMapper/ImportMap/ImportMapEntry.php

```diff
@@ -18,22 +18,75 @@
  */
 final class ImportMapEntry
 {
-    public function __construct(
-        public readonly string $importName,
-        /**
-         * The path to the asset if local or downloaded.
-         */
-        public readonly ?string $path = null,
-        public readonly ?string $version = null,
-        public readonly ImportMapType $type = ImportMapType::JS,
-        public readonly bool $isEntrypoint = false,
-        public readonly ?string $packageName = null,
-        public readonly ?string $filePath = null,
+    public readonly string $importName;
+    public readonly ImportMapType $type;
+    /**
+     * A logical path, relative path or absolute path to the file.
+     */
+    public readonly string $path;
+    public readonly bool $isEntrypoint;
+
+    // remote-only
+    public readonly string $version;
+    /**
+     * The "package-name/path" of the remote package.
+     */
+    public readonly string $packageModuleSpecifier;
+
+    private function __construct(
+        string $importName,
+        ImportMapType $type,
+        string $path,
+        bool $isEntrypoint,
+        ?string $version,
+        ?string $packageModuleSpecifier,
     ) {
+        $this->importName = $importName;
+        $this->type = $type;
+        $this->path = $path;
+        $this->isEntrypoint = $isEntrypoint;
+        if (null !== $version) {
+            $this->version = $version;
+            $this->packageModuleSpecifier = $packageModuleSpecifier;
+        }
+    }
+
+    public static function createLocal(string $importName, ImportMapType $importMapType, string $path, bool $isEntrypoint): self
+    {
+        return new self($importName, $importMapType, $path, $isEntrypoint, null, null);
+    }
+
+    public static function createRemote(string $importName, ImportMapType $importMapType, string $path, string $version, string $packageModuleSpecifier, bool $isEntrypoint): self
+    {
+        return new self($importName, $importMapType, $path, $isEntrypoint, $version, $packageModuleSpecifier);
+    }
```

Comment:
private constructor + static factories. Done this way so that IF you are remote, we can guarantee that both `$version` and `$packageModuleSpecifier` ARE included.